nixops/backends/hetzner.py

```diff
@@ -187,10 +186,12 @@ def _wait_for_rescue(self, ip):
             # so only wait for the reboot to finish when deploying real
             # systems.
             self.log_start("waiting for rescue system...")
-            dotlog = lambda: self.log_continue(".")  # NOQA
-            wait_for_tcp_port(ip, 22, open=False, callback=dotlog)
+            while self.try_ssh():
+                self.log_continue(".")
```

Comment:
All these constructs sound a bit repetitive, what about refactoring it so that we can use a single `wait_for_ssh` instead?